src/backend/marsha/core/views.py

```diff
@@ -50,22 +51,27 @@ def model(self):
     def serializer_class(self):
         """Return the serializer used by the view."""
 
-    def get_context_data(self, request):
+    def get_context_data(self):
         """Build context for template rendering of configuration data for the frontend.
 
-        Parameters
-        ----------
-        request : Request
-            passed by Django
-
         Returns
         -------
         dictionary
             context with configuration data for the frontend
 
         """
         try:
-            app_data = self._get_app_data()
+            lti = LTI(self.request, self.kwargs["uuid"])
+            lti.verify()
+
+            cache_key = "app_data|{model:s}|{domain:s}|{context:s}|{resource!s}".format(
+                model=self.model.__name__,
+                domain=lti.get_consumer_site().domain,
+                context=lti.context_id,
+                resource=lti.resource_id,
+            )
+
+            app_data = self._get_app_data(cache_key=cache_key, lti=lti)
         except (LTIException, PortabilityError) as error:
```

Comment:
So actually you should split into several try/except no? One on `lti.verify` that catches `LTIException` and one on `self._get_app_data` that catches PortabilityError?

Reply:
I can but I don't see the benefit. In both case we log the error and return the same dict.